### PR TITLE
add jenkins plugin create/delete obj test; allow for jenkins mem mon …

### PR DESCRIPTION
**Pull request by gabemontero**
…disable

@csrwng @jupierce PTAL

### CODE REVIEW

**Comment by gabemontero** [comment]
[testextended][core("jenkins-plugin test create obj delete obj")]

**Comment by csrwng** [comment]
@gabemontero is there a specific case where we want to disable the memory monitoring?

**Comment by gabemontero** [comment]
@csrwng when running on your local system, all the prints from the memory monitoring make unit testing of your new tests more cumbersome, IMO.

So left it on by default for all our ci.openshift runs, but allow for it to be turned off for local dev.

**Comment by csrwng** [comment]
@gabemontero got it, yeah makes sense

**Comment by gabemontero** [comment]
[testextended][extended:core("jenkins-plugin test create obj delete obj")]

File: test/extended/image_ecosystem/plugin.go [hunk 1]
```diff
@@ -565,42 +584,44 @@ var _ = g.Describe("[image_ecosystem][jenkins][Slow] openshift pipeline plugin",
...
-				// This is likely a temporary measure in place to extract diagnostic information during unexpectedly
-				// high memory utilization within the Jenkins image. If Jenkins is using
-				// a large amount of RSS, extract JVM information from the pod.
-				if memoryOverragePattern.MatchString(memstats) {
-					histogram, err := oc.Run("rsh").Args("--namespace", jenkinsNamespace, jenkinsPod.Name, "jmap", "-histo", "1").Output()
-					if err == nil {
-						ginkgolog("\n\nJenkins histogram:\n%s\n\n", histogram)
-					} else {
-						ginkgolog("Unable to acquire Jenkins histogram: %v", err)
+		if os.Getenv(disableJenkinsMemoryStats) == "" {
```
Comment (gabemontero) [inline test/extended/image_ecosystem/plugin.go]: The file diff isn't ideal here .... if you append `?w=1` to the url you'll see I just wrapped the ticker and go func with the `if os.Getenv...` call

Comment (csrwng) [inline test/extended/image_ecosystem/plugin.go]: ha, cool, I didn't know about `?w=1` looks pretty useful

Comment (gabemontero) [inline test/extended/image_ecosystem/plugin.go]: learned it from @jupierce who learned it from Mo :-)

**Comment by jupierce** [comment]
LGTM

**Comment by csrwng** [comment]
LGTM [merge]

**Comment by gabemontero** [comment]
A minor timing / slowness flake ... @csrwng can you repost the merge comment please?

Details:

There was one failure in the conformance tests with scale down which appears to be simply just not waiting long enough.  The test gave up on the `oc deploy --latest` outcome, with the container still in `ContainerCreating` mode, but when the test start dumping debug information, the deployment in fact had completed per inspection of the ReplicationController json:

```
Nov 22 20:14:37.953: INFO: Error running &{/data/src/github.com/openshift/origin/_output/local/bin/linux/amd64/oc [oc deploy --config=/tmp/extended-test-cli-deployment-jpa31-88s8h-user.kubeconfig --namespace=extended-test-cli-deployment-jpa31-88s8h --latest --follow deployment-test] []   Flag --latest has been deprecated, use 'oc rollout latest' instead
Started deployment #4
Error from server: container "myapp" in pod "deployment-test-4-5bquj" is waiting to start: ContainerCreating
 Flag --latest has been deprecated, use 'oc rollout latest' instead
Started deployment #4
Error from server: container "myapp" in pod "deployment-test-4-5bquj" is waiting to start: ContainerCreating
 [] <nil> 0xc8211be4c0 exit status 1 <nil> true [0xc8204fb120 0xc8204fb148 0xc8204fb148] [0xc8204fb120 0xc8204fb148] [0xc8204fb128 0xc8204fb140] [0xb1c250 0xb1c3b0] 0xc8211c0060}:
Flag --latest has been deprecated, use 'oc rollout latest' instead
Started deployment #4
Error from server: container "myapp" in pod "deployment-test-4-5bquj" is waiting to start: ContainerCreating
[AfterEach] with test deployments [Conformance]
  /data/src/github.com/openshift/origin/test/extended/deployments/deployments.go:201
Nov 22 20:14:37.954: INFO: Running 'oc get --config=/tmp/extended-test-cli-deployment-jpa31-88s8h-user.kubeconfig --namespace=extended-test-cli-deployment-jpa31-88s8h dc/deployment-test -o yaml'
Nov 22 20:14:38.449: INFO: 
apiVersion: v1
kind: DeploymentConfig
metadata:
  creationTimestamp: 2016-11-23T01:11:14Z
  generation: 6
  name: deployment-test
  namespace: extended-test-cli-deployment-jpa31-88s8h
  resourceVersion: "8348"
  selfLink: /oapi/v1/namespaces/extended-test-cli-deployment-jpa31-88s8h/deploymentconfigs/deployment-test
  uid: ba880344-b119-11e6-8b0b-0eb004199e34
spec:
  replicas: 1
  selector:
    name: deployment-test
  strategy:
    resources: {}
    rollingParams:
      intervalSeconds: 1
      maxSurge: 25%
      maxUnavailable: 25%
      pre:
        execNewPod:
          command:
          - /bin/echo
          - test pre hook executed
          containerName: myapp
        failurePolicy: Abort
      timeoutSeconds: 600
      updatePeriodSeconds: 1
    type: Rolling
  template:
    metadata:
      creationTimestamp: null
      labels:
        name: deployment-test
    spec:
      containers:
      - command:
        - /bin/sleep
        - "100"
        image: docker.io/centos:centos7
        imagePullPolicy: IfNotPresent
        name: myapp
        resources: {}
        terminationMessagePath: /dev/termination-log
      dnsPolicy: ClusterFirst
      restartPolicy: Always
      securityContext: {}
      terminationGracePeriodSeconds: 0
  test: true
  triggers:
  - type: ConfigChange
status:
  conditions:
  - lastTransitionTime: 2016-11-23T01:12:45Z
    message: Deployment config does not have minimum availability.
    status: "False"
    type: Available
  - lastTransitionTime: 2016-11-23T01:14:37Z
    message: Replication controller "deployment-test-4" has completed progressing
    reason: NewReplicationControllerAvailable
    status: "True"
    type: Progressing
  details:
    causes:
    - type: Manual
    message: manual change
  latestVersion: 4
  observedGeneration: 6
```

**Comment by csrwng** [comment]
[merge]

**Comment by gabemontero** [comment]
thx @csrwng

**Comment by csrwng** [comment]
@gabemontero if there isn't an issue for the flake, can you create one? Otherwise Jenkins won't re-test

**Comment by gabemontero** [comment]
ah .. thx @csrwng , forgot about the new process

test flake https://github.com/openshift/origin/issues/12012

**Comment by csrwng** [comment]
#12012 [merge]

**Pull request merged**